@xen-orchestra/lite/src/App/TreeView.tsx

```diff
@@ -55,9 +54,11 @@ const TreeView = withState<State, Props, Effects, Computed, ParentState, ParentE
             ?.valueSeq()
             .forEach((host: Host) => {
               const runningVms = []
-              host.resident_VMs.forEach(vmRef => {
-                let vm
-                if ((vm = state.vmsByRef?.get(vmRef)) !== undefined) {
+              state.vmsByContainerRef
+                ?.get(host.$ref)
+                ?.valueSeq()
+                .sortBy(vm => vm.name_label)
+                .forEach((vm: Vm) => {
                   runningVms.push({
```

Comment:
Why not using `.map()` instead `forEach + push` ?

Reply:
because we need an array (`children` props of `collection`) and not an immutable object.

Reply:
`.toArray()` is available on a `Map`